chain/namada/namada_chain.go

```diff
@@ -0,0 +1,1291 @@
+	if err != nil {
+		return fmt.Errorf("downloading template files failed: %v", err)
+	}
+	err = c.downloadWasms(ctx)
+	if err != nil {
+		return fmt.Errorf("downloading wasm files failed: %v", err)
+	}
+
+	err = c.setValidators(ctx)
+	if err != nil {
+		return fmt.Errorf("setting validators failed: %v", err)
+	}
+
+	err = c.initAccounts(ctx, additionalGenesisWallets...)
+	if err != nil {
+		return fmt.Errorf("initializing accounts failed: %v", err)
+	}
+
+	err = c.updateParameters(ctx)
+	if err != nil {
+		return fmt.Errorf("updating parameters failed: %v", err)
+	}
+
+	err = c.initNetwork(ctx)
+	if err != nil {
+		return fmt.Errorf("init-network failed: %v", err)
+	}
+
+	eg, egCtx := errgroup.WithContext(ctx)
+	for _, n := range c.Validators {
+		eg.Go(func() error {
+			if err := c.copyGenesisFiles(egCtx, n); err != nil {
+				return err
+			}
+			return n.CreateContainer(egCtx)
+		})
+	}
+
+	for _, n := range c.FullNodes {
+		eg.Go(func() error {
+			if err := c.copyGenesisFiles(egCtx, n); err != nil {
+				return err
+			}
+			return n.CreateContainer(egCtx)
+		})
+	}
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	eg, egCtx = errgroup.WithContext(ctx)
+	for _, n := range c.Validators {
+		eg.Go(func() error {
+			return n.StartContainer(egCtx)
+		})
+	}
+
+	for _, n := range c.FullNodes {
+		eg.Go(func() error {
+			return n.StartContainer(egCtx)
+		})
+	}
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	if err := testutil.WaitForBlocks(ctx, 2, c.Validators[0]); err != nil {
+		return err
+	}
+
+	c.isRunning = true
+
+	return nil
+}
+
+// Execute a command.
+func (c *NamadaChain) Exec(ctx context.Context, cmd []string, env []string) (stdout, stderr []byte, err error) {
+	return c.Validators[0].Exec(ctx, cmd, env)
+}
+
+// Exports the chain state at the specific height.
+func (c *NamadaChain) ExportState(ctx context.Context, height int64) (string, error) {
+	panic("implement me")
+}
+
+// Get the RPC address.
+func (c *NamadaChain) GetRPCAddress() string {
+	return fmt.Sprintf("http://%s:26657", c.Validators[0].HostName())
+}
+
+// Get the gRPC address. This isn't used for Namada.
+func (c *NamadaChain) GetGRPCAddress() string {
+	// Returns a dummy address because Namada doesn't support gRPC
+	return fmt.Sprintf("http://%s:9090", c.Validators[0].HostName())
+}
+
+// Get the host RPC address.
+func (c *NamadaChain) GetHostRPCAddress() string {
+	return "http://" + c.Validators[0].hostRPCPort
+}
+
+// Get the host peer address.
+func (c *NamadaChain) GetHostPeerAddress() string {
+	return c.Validators[0].hostP2PPort
+}
+
+// Get the host gRPC address.
+func (c *NamadaChain) GetHostGRPCAddress() string {
+	panic("No gRPC address for Namada")
+}
+
+// Get Namada home directory.
+func (c *NamadaChain) HomeDir() string {
+	return c.Validators[0].HomeDir()
+}
+
+// Create a test key.
+func (c *NamadaChain) CreateKey(ctx context.Context, keyName string) error {
+	var err error
+	cmd := []string{
+		"namadaw",
```

Comment:
I like the last line here the most if this is the convention. This way someone can swap out the namada bin (I don't expect it to happen, but want to ensure we are covered since other cosmos networks do this commonly), and still work